src/conv/invokers/flexgemm.cpp

```diff
@@ -0,0 +1,162 @@
+                      const Kernel& kern,
+                      const solver::param_conv_t& p,
+                      void* dst,
+                      const void* src)
+{
+    uint32_t bnn   = p.bnx * p.bny;
+    uint32_t lda   = p.n * bnn;
+    uint32_t align = p.id != 3 ? 7 : 15;
+    handle.Run(kern)(dst, src, lda, (p.n + 3) & ~3, p.k, p.n, bnn, (p.k + align) & ~align);
+}
+static void
+lk_genidx2d(const Handle& handle, const Kernel& kern, const solver::param_conv_t& p, void* dst)
+{
+    uint32_t npx = p.pnx * p.pny;
+    uint32_t inc = p.ng * p.inc;
+    uint32_t onc = p.ng * p.n;
+    uint32_t ldx = npx * (p.pad == 0 ? inc : 1);
+    handle.Run(kern)(
+        dst, p.ntidx, p.pnx, p.sd, ldx, onc, p.m, p.cnx, p.cny, p.bnx, p.bny, p.lda, p.k);
+}
+static void lk_conv(const Handle& handle,
+                    const Kernel& kern,
+                    const solver::param_conv_t& p,
+                    void* c,
+                    const void* a,
+                    const void* b,
+                    const void* idx,
+                    float alpha)
+{
+    const uint8_t* relo = static_cast<const uint8_t*>(idx) + (p.ntidx << 3);
+    uint32_t n          = (p.pad != 0 ? 0x80000000 : 0) | p.n;
+    if(p.dir == 0)
+    {
+        handle.Run(kern)(idx, relo, n, p.k, p.ags, alpha, a, b, c, p.m, p.ldc);
+    }
+    else
+    {
+        uint32_t align = p.id != 3 ? 7 : 15;
+        uint32_t k     = (p.k + align) & ~align;
+        handle.Run(kern)(idx, relo, (p.n + 3) & ~3, n, k, p.ags, a, b, c, alpha, p.m, p.ldc);
+    }
+}
+
+namespace conv {
+InvokerFactory MakeFlexgemmInvokerFactory(const solver::param_ufconv_t& p, float alpha)
+{
+    return [=](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle, const boost::any& prim_params) {
+            const auto& tensors = boost::any_cast<DataInvokeParams>(prim_params).tensors;
+            lk_ufconv(handle, kernels[0], p, tensors.out, tensors.in, tensors.w, alpha);
+            if(handle.IsProfilingEnabled())
+            {
+                float elapsed = handle.GetKernelTime();
+                handle.ResetKernelTime();
+                handle.AccumKernelTime(elapsed);
+            }
+        };
+    };
+}
+InvokerFactory MakeFlexgemmInvokerFactory(const solver::param_conv_t& p, float alpha)
+{
+    return [=](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle, const boost::any& prim_params) {
+            const auto& params   = boost::any_cast<DataInvokeParams>(prim_params);
+            const size_t auxsize = get_auxbuf_size(p);
+            if(params.workSpace == nullptr || params.workSpaceSize < auxsize)
+                MIOPEN_THROW("Workspace is not enough for flexgemm");
+            const auto& tensors = params.tensors;
+            float elapsed       = 0.f;
+            uint8_t* a          = reinterpret_cast<uint8_t*>(params.workSpace);
+            uint8_t* b          = a + p.spad;
+            uint8_t* idx        = b + p.sperm;
```

Comment:
Can we rename it to s_perm or maybe even something else? I am talking about `p.sperm` member.

Reply:
Yes